src/Components/Artwork/__tests__/Details.test.tsx

```diff
@@ -30,75 +28,123 @@ describe("Details", () => {
-      // FIXME: These type failures should be fixed by creating a new object,
-      //        rather than mutating. E.g.
-      //
-      // const data = {
-      //   ...artworkInAuction,
-      //   sale: { ...artworkInAuction.sale, is_closed: true },
-      // }
-      const data = cloneDeep(artworkInAuction)
-      data.sale.is_closed = true
-      const wrapper = await getWrapper(data)
-      expect(wrapper.html()).toContain("Bidding closed")
-    })
-
     it("shows highest bid if sale open and highest bid", async () => {
-      const data = cloneDeep(artworkInAuction)
-      const wrapper = await getWrapper(data)
+      const wrapper = await getWrapper(artworkInAuction)
       const html = wrapper.html()
       expect(html).toContain("$2,600")
     })
 
+    it("shows 'bidding closed' message if in closed auction", async () => {
+      const data = {
+        ...artworkInAuction,
+        sale: { ...artworkInAuction.sale, is_closed: true },
+      }
+      const wrapper = await getWrapper(data)
+      expect(wrapper.html()).toContain("Bidding closed")
+    })
+
     it("shows opening bid if sale open and no highest bid", async () => {
-      const data = cloneDeep(artworkInAuction)
-      data.sale_artwork.highest_bid.display = null
+      const data = {
+        ...artworkInAuction,
+        sale_artwork: {
+          ...artworkInAuction.sale_artwork,
+          highest_bid: {
+            ...artworkInAuction.sale_artwork.highest_bid,
+            display: null,
+          },
+        },
+      }
       const wrapper = await getWrapper(data)
       const html = wrapper.html()
       expect(html).toContain("$2,400")
     })
 
     it("shows Contact for price if sale_message equals the same", async () => {
-      const data = cloneDeep(artworkInAuction)
-      data.sale.is_auction = false
-      data.sale_message = "Contact For Price"
+      const data = {
+        ...artworkInAuction,
+        sale: {
+          ...artworkInAuction.sale,
+          is_auction: false,
+          sale_message: "Contact For Price",
+        },
+      }
       const wrapper = await getWrapper(data)
       const html = wrapper.html()
       expect(html).toContain("Contact for price")
     })
 
     it("shows sale message if sale open and no bids", async () => {
-      const data = cloneDeep(artworkInAuction)
-      data.sale_artwork.highest_bid.display = null
-      data.sale_artwork.opening_bid.display = null
-      data.sale.is_auction = false
+      const data = {
+        ...artworkInAuction,
+        sale: {
+          ...artworkInAuction.sale,
+          is_auction: false,
+        },
+        sale_artwork: {
+          ...artworkInAuction.sale_artwork,
+          highest_bid: {
+            ...artworkInAuction.sale_artwork.highest_bid,
+            display: null,
+          },
+          opening_bid: {
+            ...artworkInAuction.sale_artwork.highest_bid,
+            display: null,
+          },
+        },
+      }
```

Comment:
Is this necessary?

Reply:
Probably not, but wasn’t going to spend more time than necessary during the migration phase.